debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -93,8 +100,14 @@ public EventDispatcher(CommonConnectorConfig connectorConfig, TopicSelector<T> t
         this.inconsistentSchemaHandler = inconsistentSchemaHandler != null ? inconsistentSchemaHandler : this::errorOnMissingSchema;
 
         this.transactionMonitor = new TransactionMonitor(connectorConfig, metadataProvider, this::dispatchTransactionMessage);
-        heartbeat = Heartbeat.create(connectorConfig.getConfig(), topicSelector.getHeartbeatTopic(),
-                connectorConfig.getLogicalName());
+        if (customHeartbeat != null) {
+            heartbeat = customHeartbeat;
+        }
+        else {
+            heartbeat = Heartbeat.create(connectorConfig.getConfig(), topicSelector.getHeartbeatTopic(),
```

Comment:
Would it be possible to create another factory method in `heartbeat` that would accept the additional JdbcConnection parameter so it would return the DatabaseHeartbeatImpl.

Reply:
done